src/Microsoft.DocAsCode.HtmlToPdf/HtmlToPdfConverter.cs

```diff
@@ -209,36 +211,31 @@ private void CreateOutlines(Dictionary<string, object> rootOutline, IList<HtmlMo
                         }
                     }
 
-                    ((List<Dictionary<string, object>>)rootOutline[OutLineKidsName]).Add(outline);
-                    CreateOutlines(outline, htmlModel.Children, pdfFileNumberOfPages);
+                    outlineCollection.Add(outline);
+                    CreateOutlines(outline.Outlines, htmlModel.Children, pdfFileNumberOfPages);
                 }
             }
         }
 
-        private List<Dictionary<string, object>> ConvertOutlines()
+        private void AddOutlines(PdfDocument pdfDocument)
         {
             var pdfFileNumberOfPages = GetHtmlToPdfNumberOfPages(new List<string>(_htmlFilePaths));
             _currentNumberOfPages = 1;
 
-            var rootOutline = new Dictionary<string, object>
-            {
-                { OutLineKidsName, new List<Dictionary<string, object>>() }
-            };
-
-            CreateOutlines(rootOutline, _htmlModels, pdfFileNumberOfPages);
-            return (List<Dictionary<string, object>>)rootOutline[OutLineKidsName];
+            CreateOutlines(pdfDocument.Outlines, _htmlModels, pdfFileNumberOfPages);
         }
 
-        private IList<Dictionary<string, object>> GetOutlines()
+        private void CreateOutlines(PdfDocument pdfDocument)
         {
             switch (_htmlToPdfOptions.OutlineOption)
             {
                 case OutlineOption.CustomOutline:
-                    return CustomOutlines;
+                    throw new NotImplementedException();
```

Comment:
Does this mean `OutlineOption.CustomOutline` will no longer be supported after using `PDFSharp`? Or will you implement this later?

Reply:
How is it currently being used?

Reply:
I have no knowledge about this... @partychen Do you know this?

Of course, this is not a major issue. I will merge this PR first. Thank you for contribution!